### PR TITLE
Unbuild

### PR DESCRIPTION
**Pull request by gabbaxx**
First iteration of the unbuild command!

### CODE REVIEW

File: cmd/unbuild.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	"github.com/qlik-oss/corectl/internal"
+	"github.com/spf13/cobra"
+)
+
+var unbuildCmd = withLocalFlags(&cobra.Command{
+	Use:     "unbuild",
+	Args:    cobra.ExactArgs(0),
+	Short:   "Split upp an existing app into separate entities",
+	Long:    "Split upp an existing app into separate entities",
+	Example: `corectl unbuild`,
```
Comment (wennmo) [inline cmd/unbuild.go]: We should add an `x-qlik-stability` to `experimental` on this command.

File: cmd/unbuild.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	Annotations: map[string]string{
+		"command_category": "build",
+	},
+
+	Run: func(ccmd *cobra.Command, args []string) {
+		ctx := rootCtx
+		outdir := ccmd.Flag("dir").Value.String()
+		state := internal.PrepareEngineState(ctx, headers, false)
```
Comment (wennmo) [inline cmd/unbuild.go]: Maybe we should always open without data?

File: cmd/unbuild.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+		if outdir == DefaultUnbuild {
+			appLayout, _ := state.Doc.GetAppLayout(ctx)
+			if appLayout.Title != "" {
+				outdir = internal.BuildRootFolderFromTitle(appLayout.Title)
+			} else {
+				outdir = "unknown-app-unbuild"
```
Comment (wennmo) [inline cmd/unbuild.go]: Not sure how common it is to have an app without a title, but maybe we should use the app id/name instead if title is missing?

File: cmd/unbuild.go [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+import (
+	"github.com/qlik-oss/corectl/internal"
+	"github.com/spf13/cobra"
+)
+
+var unbuildCmd = withLocalFlags(&cobra.Command{
+	Use:     "unbuild",
+	Args:    cobra.ExactArgs(0),
+	Short:   "Split upp an existing app into separate entities",
+	Long:    "Split upp an existing app into separate entities",
```
Comment (wennmo) [inline cmd/unbuild.go]: We should probably add more info to the `Long` description, e.g. what entities we do export.

File: docs/spec.json [hunk 5]
```diff
@@ -364,6 +364,15 @@
     "tables": {
       "description": "Print tables for the data model in an app"
     },
+    "unbuild": {
+      "description": "Split upp an existing app into separate entities",
+      "flags": {
+        "dir": {
+          "description": "Path to a the folder where the unbuilt app is exported",
+          "default": "./\u003capp name\u003e-unbuild"
```
Comment (wennmo) [inline docs/spec.json]: We need to check with regards to the spec definition what should be present here

File: cmd/unbuild.go [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+	Example: `corectl unbuild`,
+	Annotations: map[string]string{
+		"command_category": "build",
+		"x-qlik-stability": "experimental",
+	},
+
+	Run: func(ccmd *cobra.Command, args []string) {
+		ctx := rootCtx
+		viper.Set("no-data", "true") // Force no-data since we only use metadata
+		outdir := ccmd.Flag("dir").Value.String()
```
Comment (glooms) [inline cmd/unbuild.go]: Admittedly very tiny, but maybe `outdir` should be `outDir`?
Otherways, really good work!

**Pull request merged**